File: 백준/Silver/19637. IF문 좀 대신 써줘/IF문 좀 대신 써줘.py
n, caseN = list(map(int, input().split()))
arr = []
test = []

for i in range(n):
  s,v = input().split()
  arr.append([s, int(v)])

for i in range(caseN):
  test.append(int(input()))

answer = []

for val in test:
  lt = 0
  rt = n -1
  index = float("inf")

  while lt <= rt:
    mid = (lt + rt) //2

    if arr[mid][1] >= val:
      rt = mid - 1
      index = min(index, mid)
    else:
      lt = mid + 1

  answer.append(arr[index][0])

print("\n".join(answer))

File: 백준/Silver/19637. IF문 좀 대신 써줘/IF문 좀 대신 써줘.js
const input = require("fs")
  .readFileSync("/dev/stdin", "utf8")
  .trim()
  .split("\n");

const solution = (input) => {
  const [n, caseN] = input[0].split(" ").map((v) => Number(v));
  const arr = [];
  const test = [];

  for (let i = 1; i <= n; i++) {
    const a = input[i].split(" ");
    arr.push([a[0], Number(a[1])]);
  }
  for (let i = n + 1; i < n + caseN + 1; i++) {
    test.push(Number(input[i]));
  }

  const answer = [];

  for (const val of test) {
    let lt = 0;
    let rt = n - 1;
    let index = Infinity;

    while (lt <= rt) {
      const mid = Math.floor((lt + rt) / 2);

      if (arr[mid][1] >= val) {
        rt = mid - 1;
        index = Math.min(index, mid);
      } else {
        lt = mid + 1;
      }
    }

    answer.push(arr[index][0]);
  }

  console.log(answer.join("\n"));
};

solution(input);
